tests/rados/pool_tests.py

```diff
@@ -565,6 +565,122 @@ def run(ceph_cluster, **kw):
+        rados_obj.create_pool(pool_name=pool_name)
+        rados_obj.bench_write(pool_name=pool_name)
+
+        log.info("Getting a sample PG from the pool")
+        pool_id = pool_obj.get_pool_id(pool_name=pool_name)
+        pgid = f"{pool_id}.0"
+        pg_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+        log.info(f"Pg set fetched from pool id {pool_id} for pg id {pgid} is {pg_set}")
+        osd_tree_op = rados_obj.run_ceph_command(cmd="ceph osd tree")
+        log.info(
+            f"Before bringing down osds, output from ceph osd tree : {osd_tree_op}"
+        )
+
+        osd_pick = pg_set.pop()
+        second_osd_pick = pg_set.pop()
+        log.info(
+            f"Bringing down two OSDs from the acting set : {osd_pick} and {second_osd_pick}"
+        )
+        if not rados_obj.change_osd_state(action="stop", target=osd_pick, timeout=20):
+            log.error(f"Failed to stop OSD : {osd_pick}")
+            raise Exception("test-bed set-up failure")
+        time.sleep(10)
+
+        if not rados_obj.change_osd_state(
+            action="stop", target=second_osd_pick, timeout=20
+        ):
+            log.error(f"Failed to stop OSD : {osd_pick}")
+            raise Exception("test-bed set-up failure")
+
+        # Get the min_size property of the pool
+        min_size_val = 0
+        min_size_val = rados_obj.get_pool_property(pool=pool_name, props="min_size")
+        log.info(f"Current min_size of the pool - {min_size_val}")
+
+        log.info("Performing rados put op")
+        client_node = ceph_cluster.get_nodes(role="client")[0]
+        pool_stat = rados_obj.get_cephdf_stats(pool_name=pool_name)
+        log.debug(pool_stat)
+        total_objs = pool_stat["stats"]["objects"]
+        pool_obj.do_rados_put(client=client_node, pool=pool_name, nobj=10, timeout=10)
+        pool_stat_post = rados_obj.get_cephdf_stats(pool_name=pool_name)
+        log.debug(pool_stat_post)
+        if pool_stat_post["stats"]["objects"] != total_objs:
+            log.error(
+                "Write ops should not have been possible, number of objects in the pool have changed"
+            )
+            raise Exception(
+                f"Pool {pool_name} has {pool_stat_post['stats']['objects']} objs | Expected {total_objs} objs"
+            )
+        else:
+            log.info(
+                "RADOS PUT op failed as expected since cluster is in degraded state!! "
+            )
+
+        # Setting the min-size as 1. The IOs must be resumed and the PGs should become active+clean state.
+        if not rados_obj.set_pool_property(pool=pool_name, props="min_size", value=1):
+            log.error(f"failed to set min_size 1 on pool {pool_name}")
+            raise Exception("Config set failed on pool")
+
+        min_size_val = rados_obj.get_pool_property(pool=pool_name, props="min_size")
+        log.info(f"min_size of the pool after modification - {min_size_val}")
+
+        pg_set = rados_obj.get_pg_acting_set(pg_num=pgid)
```

Comment:
Is it necessary to fetch the `pg acting set` for the pg id again?

Reply:
Please compare the acting pg set post modification with the original acting set of the pgid and ensure that new acting pg set has only one OSD and it is the one which was not taken down.